salt/modules/firewalld.py

```diff
@@ -335,10 +331,13 @@ def list_services(zone=None):
     else:
         cmd = '--list-services'
 
+    if permanent:
+        cmd += ' --permanent'
+
     return __firewall_cmd(cmd).split()
 
 
-def add_service(name, zone=None, permanent=True):
+def add_service(zone, service, permanent=True):
     '''
     Add a service for zone. If zone is omitted, default zone will be used.
```

Comment:
This comment needs to be updated, because `zone` cannot be omitted anymore. Same with `remove_service()` further down. Why did you choose to swap the parameters?